### ISSUE TITLE
Crash on Android 6 or lower - java.lang.NoSuchMethodError: No interface method putIfAbsent

### ISSUE THREAD

**Issue by mnaturel** [GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk#124]
**Describe the bug**

When calling `Recaptcha.fetchClient` or `Recaptcha.getClient` on devices running on Android 6 or lower, there is an error due to usage of `java.util.Map.putIfAbsent` which is only available starting from API 24.

It seems it has been introduced in the 18.6.0 SDK version (e.g. no error in the 18.5.1 version)

Stacktrace of the error:

```
java.lang.NoSuchMethodError: No interface method putIfAbsent(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object; in class Ljava/util/Map; or its super classes (declaration of 'java.util.Map' appears in /system/framework/core-libart.jar)
                                                                                                    	at com.google.android.recaptcha.internal.zzcx.<init>(com.google.android.recaptcha:recaptcha@@18.6.0:30)
                                                                                                    	at com.google.android.recaptcha.internal.zzcs.zza(com.google.android.recaptcha:recaptcha@@18.6.0:1)
                                                                                                    	at com.google.android.recaptcha.internal.zzcs.zzb(com.google.android.recaptcha:recaptcha@@18.6.0:1)
                                                                                                    	at com.google.android.recaptcha.Recaptcha.getClient-BWLJW6A(com.google.android.recaptcha:recaptcha@@18.6.0:3)
                                                                                                    	at com.google.android.recaptcha.Recaptcha.getClient-BWLJW6A$default(com.google.android.recaptcha:recaptcha@@18.6.0:1)
```

**Integration Method**

Select the method used to integrate with reCAPTCHA Mobile.

* [ ] SPM (iOS)
* [ ] Cocoapods (iOS)
* [ ] Direct Download (iOS)
* [x] GMaven (Android)

**SDK Version (e.g. 18.0.1):** 18.6.0

**To Reproduce**

1. Use a device running Android 6 or lower
2. Try to get the Recaptcha client

**Expected behavior**

The client should be initialized without any error even on device running Android 6 or lower.

**Screenshots**

N/A

**Xcode version for iOS (please complete the following information):**
 - Version [e.g. Xcode 13.4.1] N/A

**Device (please complete the following information):**
 - Model: [e.g. Pixel 7, iPhone6] Pixel 5
 - OS: [e.g. Android 28, iOS8.1] Android 6 (API 23)
 - Emulator/Simulator or real Device?  Emulator

**Additional context**

N/A

**Comment by walterjgsp** [comment]
Thanks for the report @mnaturel. We will take a look internally and fix it as soon as possible.

**Comment by walterjgsp** [comment]
@mnaturel, we released the new 18.6.1 that should have this fixed. Please let me know if there's anything else still ongoing related to this in the new version.

**Comment by mnaturel** [comment]
Thanks @walterjgsp, the 18.6.1 version fixes the issue.
